engine.go

```diff
@@ -51,8 +52,9 @@ type ColumnWithRawDefault struct {
 }
 
 // New creates a new Engine with custom configuration. To create an Engine with
-// the default settings use `NewDefault`.
-func New(a *analyzer.Analyzer, cfg *Config) *Engine {
+// the default settings use `NewDefault`. Should call Engine.Close() to finalize
+// dependency lifecycles.
+func New(a *analyzer.Analyzer, cfg *Config, bThreads *sql.BackgroundThreads) *Engine {
```

Comment:
Should not take this new param, just construct it yourself.

The key is that this is an engine-managed construct that integrators add things to, no reason to allow it to be configurable like this.